### DISCUSSION TITLE
Need help in upgrading netbox from 2.7.11 to next version

### DISCUSSION THREAD

**Discussion by vbRaghu** [netbox-community/netbox#15592] [Help Wanted!]
Hi Team,
We are using netbox 2.7.11 on CentOS Linux release 8.1.1911 
[root@netb]# python3 -V
Python 3.6.8
[root@netb]# find / -name postgres
/usr/bin/postgres
/usr/pgsql-9.6/bin/postgres
[root@netb]# redis-server --version
Redis server v=5.0.3 sha=00000000:0 malloc=jemalloc-5.1.0 bits=64 build=8c0bf22bfba82c8f

I would like to know to which version we can proceed with upgrade. Since I saw in the net that we can perform only sequential upgrade so I would to know which version we can perform next. Also I need to steps to perform the upgrade

**Comment by candlerb** [comment]
From a Netbox point of view, you need to upgrade from 2.7.11 to 2.11.12 as the first step, then to 3.7.4 as the second step. (There are no sequencing requirements except that you must upgrade to 2.11.x before you upgrade to 3.x)

However, you also need to ensure you have the correct dependencies.
* For Netbox 2.11 the [requirements](https://github.com/netbox-community/netbox/blob/v2.11.12/docs/installation/upgrading.md) are python 3.6-3.8 and Postgres 9.6, so you should be OK
* For the latest Netbox the [requirements](https://docs.netbox.dev/en/stable/installation/upgrading/#2-update-dependencies-to-required-versions) are python 3.8+ and postgres 12+

However, for such a large upgrade I strongly recommend you do this using completely fresh VMs rather than in-place upgrades.
* Firstly, make a fresh VM of 2.11.12 and test it works (e.g. create a few sample devices).
* pgdump the SQL data from your 2.7.11 instance
* Import the pgdump into your 2.11.12 instance. Run `./upgrade.sh` to migrate it, then check it works
* Make another fresh VM, this time of 3.7.4, and test it
* pgdump the SQL data from your 2.11.12 instance
* Import the pgdump into your 3.7.4 instance. Run `./upgrade.sh` to migrate it, then check it works
* That's your new production instance

There have been large changes to the Netbox data models and APIs, so any existing integrations you have (e.g. to the REST API) will need to be modified to work with the new version. Read the release notes for all intervening vX.X.0 versions and look for "Breaking changes"

If you want to do an "in place" upgrade then you do this at your own risk - make sure you've [backed up](https://docs.netbox.dev/en/stable/administration/replicating-netbox/) your Postgres database first - although an in-place upgrade to 2.11.12 is relatively low risk. The upgrade steps are documented [here](https://docs.netbox.dev/en/stable/installation/upgrading/). Be careful to use the correct set of steps, depending on whether the installation was originally from git or from a tarball. (Using `ls -l /opt/netbox/.git` and `ls -l /opt/netbox` will tell you which).

Check that your existing installation is is using a virtualenv (this was introduced in 2.7.9 I believe) and running from systemd rather than supervisord (I'm not sure when that change was made).  If either of those is not the case then it makes an in-place upgrade more difficult.

**Reply by vbRaghu** [reply]
Hi Brain,

Thank you for your help. I will proceed with creating new VM RHEL8/9 and then will upgrade.
Could you please let me know how to export and import the pgdump which are running in netbox version 2.7.11 into netbox version 2.11.12.

Please provide with the steps so that I can proceed with upgrade.

Thanks once again.

**Reply by candlerb** [reply]
https://docs.netbox.dev/en/stable/administration/replicating-netbox/

**Comment by vbRaghu** [comment]
thank you so much for your guidelines, upgraded the netbox to 3.2.9
